modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectFieldLocalServiceImpl.java

```diff
@@ -308,6 +322,51 @@ private ObjectField _addObjectField(
 		return objectFieldPersistence.update(objectField);
 	}
 
+	private void _addObjectFieldSettings(
+			long userId, long objectFieldId,
+			List<ObjectFieldSetting> newObjectFieldSettings)
+		throws PortalException {
+
+		List<ObjectFieldSetting> oldObjectFieldSettings =
+			_objectFieldSettingPersistence.findByObjectFieldId(objectFieldId);
+
+		for (ObjectFieldSetting oldObjectFieldSetting :
+				oldObjectFieldSettings) {
+
+			Stream<ObjectFieldSetting> stream = newObjectFieldSettings.stream();
+
+			Optional<ObjectFieldSetting> optional = stream.filter(
+				newObjectFieldSetting ->
+					newObjectFieldSetting.getName() ==
+						oldObjectFieldSetting.getName()
+			).findFirst();
+
+			if (!optional.isPresent()) {
+				_objectFieldSettingPersistence.remove(oldObjectFieldSetting);
+			}
+		}
+
+		for (ObjectFieldSetting newObjectFieldSetting :
+				newObjectFieldSettings) {
+
+			ObjectFieldSetting oldObjectFieldSetting =
+				_objectFieldSettingPersistence.fetchByOFI_N(
+					objectFieldId, newObjectFieldSetting.getName());
+
+			if (oldObjectFieldSetting == null) {
+				_objectFieldSettingLocalService.addObjectFieldSetting(
+					userId, objectFieldId, newObjectFieldSetting.getName(),
+					newObjectFieldSetting.isRequired(),
+					newObjectFieldSetting.getValue());
+			}
+			else {
+				_objectFieldSettingLocalService.updateObjectFieldSetting(
```

Comment:
I noticed the required attribute is not being updated. Can't that happen?

Reply:
Considering this story, it can't. It should always be the same, that's why I didn't include it in the update.